netket/nn/symmetric_linear.py

```diff
@@ -211,35 +572,216 @@ def full_bias(self, bias):
+        dtype = jnp.promote_types(x.dtype, self.dtype)
+        x = jnp.asarray(x, dtype)
+
+        x = x.reshape(-1, x.shape[1] * x.shape[2])
 
         kernel = self.param(
             "kernel",
             self.kernel_init,
-            (inputs.shape[-1] // self.in_features, self.in_features, self.out_features),
+            (self.out_features, self.in_features, self.n_symm),
             self.dtype,
         )
+
+        kernel = jnp.asarray(kernel, dtype)
+
+        if self.mask:
+            kernel = kernel * jnp.expand_dims(self.mask, (0, 1))
+
         kernel = self.full_kernel(kernel)
         kernel = jnp.asarray(kernel, dtype)
 
-        y = lax.dot_general(
-            inputs,
+        x = lax.dot_general(
+            x,
             kernel,
-            (((inputs.ndim - 1,), (0,)), ((), ())),
+            (((x.ndim - 1,), (0,)), ((), ())),
             precision=self.precision,
         )
 
         if self.use_bias:
             bias = self.param("bias", self.bias_init, (self.out_features,), self.dtype)
             bias = jnp.asarray(self.full_bias(bias), dtype)
-            y += bias
+            x += bias
+
+        return x
+
+
+def DenseSymm(symmetries, mode="auto", shape=None, point_group=None, **kwargs):
+    """
+    Implements a projection onto a symmetry group. The output will be
+    equivariant with respect to the symmetry operations in the group and can
+    be averaged to produce an invariant model.
+
+    Args:
+        symmetries: A specification of the symmetry group. Can be given by a
+            nk.graph.Graph, a nk.utils.PermuationGroup, or an array [n_symm, n_sites]
+            specifying the permutations corresponding to symmetry transformations
+            of the lattice.
+        point_group: If symmetries is a graph the default point group of the
+            graph is overwritten from which the space group is build. To only
+            symmetrize over the point_group instead set symmetries=point_group
+        mode: string "fft, matrix, auto" specifying whether to use a fast Fourier
+            transform, matrix multiplication, or to choose a sensible default
+            based on the symmetry group
+        features: The number of symmetry-reduced features. The full output size
+            is [n_symm,features].
+        use_bias: A bool specifying whether to add a bias to the output (default: True).
+        mask: An optional array of shape [n_sites] consisting of ones and zeros
+            that can be used to give the kernel a particular shape
+        dtype: The datatype of the weights. Defaults to a 64bit float
+        precision: Optional argument specifying numerical precision of the computation
+            see `jax.lax.Precision`for details.
+        kernel_init: Optional kernel initialization function. Defaults to variance scaling
+        bias_init: Optional bias initialization function. Defaults to zero initialization
+    """
+
+    if isinstance(symmetries, Graph):
```

Comment:
shouldn't you be checking for Lattice? a Graph does not have space_group.

```suggestion
    if isinstance(symmetries, Lattice):
```

Reply:
Actually Graph is correct here because we look for a point group to construct the space group and if we don't find one we call automorphisms() which is not restricted to lattices.

Reply:
then the correct implementation, in my mind, would be

```python
if isinstance Lattice:
    # the line below works both with point_group=None or not
    sym = HashableArray(np.asarray(symmetries.space_group(point_group)))
    mode = "fft" if mode == "auto" ... 
elif isinstance Graph:
    sym = HashableArray(np.asarray(symmetries.automorphisms()))
  ....
```

Reply:
the point group stuff is specific to Lattices.

Reply:
Yes, you are correct. Also looking for _point_group in a Graph may raise an error so I'll fix this